File: .NET/Survey/TabOrderHelper/Form1.cs
using System;
using System.Windows.Forms;

namespace TabOrderHelper
{
    public partial class Form1 : Form
    {
        private TabOrderHelper _helper = null;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            _helper = new TabOrderHelper(this);
        }

        //protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
        //{
        //    var activeControl = this.ActiveControl;

        //    Console.WriteLine($"activeControl={activeControl.Name}");

        //    if (keyData == Keys.Tab)
        //    {
        //        // TABキーが押されたときの処理
        //        //System.Diagnostics.Debug.WriteLine("TAB key pressed");

        //        var nextControl = _helper.GetNextControl(activeControl);

        //        nextControl.Focus();

        //        return true; // イベントを処理済みとしてマークする
        //    }
        //    else if (keyData == (Keys.Shift | Keys.Tab))
        //    {
        //        // SHIFT+TABキーが押されたときの処理
        //        //System.Diagnostics.Debug.WriteLine("SHIFT+TAB key pressed");

        //        var prevControl = _helper.GetNextControl(activeControl, false);

        //        prevControl.Focus();

        //        return true; // イベントを処理済みとしてマークする
        //    }

        //    return base.ProcessCmdKey(ref msg, keyData);
        //}
    }
}

File: .NET/Survey/TabOrderHelperVB/Form1.vb
Public Class Form1
    Private _helper As TabOrderHelper.TabOrderHelper = Nothing

    Private Sub Form1_Load(sender As Object, e As EventArgs)
        _helper = New TabOrderHelper.TabOrderHelper(Me)
    End Sub

    Protected Overrides Function ProcessCmdKey(ByRef msg As Message, ByVal keyData As Keys) As Boolean
        Dim control = Me.ActiveControl

        If keyData = Keys.Tab Then
            ' TABキーが押されたときの処理

            Dim nextControl = _helper.GetNextControl(control)
            nextControl.Focus()
            Return True ' イベントを処理済みとしてマークする

        ElseIf keyData = (Keys.Shift Or Keys.Tab) Then
            ' SHIFT+TABキーが押されたときの処理

            Dim prevControl = _helper.GetNextControl(control, False)
            prevControl.Focus()
            Return True ' イベントを処理済みとしてマークする

        End If

        Return MyBase.ProcessCmdKey(msg, keyData)
    End Function

End Class